raft/status.go

```diff
@@ -47,3 +49,20 @@ func getStatus(r *raft) Status {
 
 	return s
 }
+
+func (s Status) MarshalJSON() ([]byte, error) {
+	j := fmt.Sprintf(`{"id":"%x","term":%d,"vote":"%x","commit":%d,"lead":"%x","raftState":"%s","progress":{`,
```

Comment:
If we defined a `type NodeID uint64` we could implement MarshalJSON on the NodeID type to print in hex. Would that be enough to avoid the more complicated MarshalJSON override here?  Changing uint64 to NodeID would touch a lot of code but it would be largely mechanical (and gogoproto even lets you use custom types inside proto objects)